### ISSUE TITLE
Proposal: Add isUseCaseFinished to DispatcherPayloadMeta

### ISSUE THREAD

**Issue by azu** [almin/almin#149]
I want to add `isUseCaseFinished` to `DispatcherPayloadMeta` class.

## Motivation

I want to know if the UseCase is finished.
Of course, We can know this by using `Context#onCompleteEachUseCase`.
But, `Context#onCompleteEachUseCase(handler)` is called by async.

- [LifeCycle of UseCase · Almin.js](https://almin.js.org/docs/tips/usecase-lifecycle.html  "LifeCycle of UseCase · Almin.js")

### use-case

I want to do task at some `UseCase` is finished.

Currently, we do that by following steps.

- Listen `onDidExecuteEachUseCase` adn dos the task
- Listen `onCompleteEachUseCase` and do the task

```js
// same with `onCompleteEachUseCase` and `onDidExecuteEachUseCase`
dispatcher.onDispatch((payload: Payload) => {
  if (payload instanceof DidExecutedPayload
  || payload instanceof CompletedPayload) {
  	// do task
  }
});
```

The above way has a problem.
When the `UseCase` is finished, the task is ran **twice**.

I want to do the task only once.

## After changed

We can write this as following:

```js
// same with `onCompleteEachUseCase` and `onDidExecuteEachUseCase`
const finishedCallback = () => {
    // do task
};
const calledMap = new Map();
dispatcher.onDispatch((payload, meta) => {
    if (payload instanceof DidExecutedPayload && meta.isUseCaseFinished) {
        calledMap.set(meta.useCase, true);
        finishedCallback();
    } else if (payload instanceof CompletedPayload && meta.isUseCaseFinished) {
        if (calledMap.has(meta.useCase)) {
            return void calledMap.delete(meta.useCase);
        }
        finishedCallback();
    }
});
```

**Comment by azu** [comment]
[![flow chat](https://code2flow.com/94rfp0.png)](https://code2flow.com/94rfp0)

**Comment by azu** [comment]
Implemented in #150
